<launch>

    <!-- Sensors -->
    
    <!-- IMU -->
    <include file="$(find xsens_mti_driver)/launch/xsens_mti_node.launch" >
        <arg name="port" value="/dev/ttyUSB1" />
    </include>
    
    <!-- LiDAR -->
    <include file="$(find velodyne_pointcloud)/launch/VLP16_points.launch" />
    <node pkg="lms1xx" type="LMS1xx_node" name="lms1xx" output="screen">
        <param name="sensor_param" value="param_value" />
    </node>

    <node pkg="laser_filters" type="scan_to_scan_filter_chain" name="laser_filter">
        <rosparam command="load" file="$(find laser_filters)/my_laser_config.yaml" />
    </node>

    <node pkg="tf" type="static_transform_publisher" name="laser_tf_publisher" args="0.57 0.0 0.30 0 0.0 0.0 base_link laser 10" />

    <!-- Sensors end -->
    
    <!-- Odometry -->
    <param name="erp_port" type="str" value="/dev/ttyUSB0" />
    <node name="erp42_encoder" pkg="odometry" type="erp42_encoder.py" output="screen" />
    <node name="erp42_odometry" pkg="odometry" type="erp42_odometry.py" />
    <!-- <node name="fake_odometry" pkg="odometry" type="fakeOdometry.py" output="screen" /> -->

    
    <!-- Path Publish --> 
    <param name="c_gain" type="double" value="1.2" />
    <param name="desired_speed" type="double" value="8.0" />
    <node name="stanley_method" pkg="path_planner" type="global_stanley.py" output="screen" />

    <!-- Obstacle -->
    <node pkg="cone_tracker" type="cone.py" name="cone" output="screen"/>
    <node pkg="cone_tracker" type="transform_laser_to_odom.py" name="transform_laser_to_odom"/>    

    <!-- Localization -->
    <include file="$(find hdl_localization)/launch/hdl_localization.launch">
        <arg name="map_file" value="$(find hdl_localization)/data/Globalmap2.pcd" /> 
    </include>

    <!-- Model -->
    <include file="$(find erp42_vehicle_rviz)/launch/erp42_vehicle_rviz.launch" />

    <!-- Test -->
    <!-- <node pkg="tf" type="static_transform_publisher" name="map_to_odom" args="-2.67 -3.75 0.0 -2.29 0.0 0.0 map odom 10" /> -->
    <!-- -2.676659107208252,-3.7524757385253906,-2.29648101519 -->


</launch>